x-pack/filebeat/input/netflow/decoder/v9/lru_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package v9
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPendingTemplatesCache(t *testing.T) {
+
+	type testEvent struct {
+		key SessionKey
+		buf *bytes.Buffer
+	}
+
+	tests := []struct {
+		name             string
+		eventsToAdd      []testEvent
+		eventsToGet      []SessionKey
+		eventsExpected   []*bytes.Buffer
+		getDelay         time.Duration
+		cleanInterval    time.Duration
+		removalThreshold time.Duration
+	}{
+		{
+			name: "Add and GetAndRemove different sessions with cache hit",
+			eventsToAdd: []testEvent{
+				{SessionKey{"127.0.0.1", 0}, bytes.NewBufferString("test-event-1")},
+				{SessionKey{"127.0.0.2", 0}, bytes.NewBufferString("test-event-1")},
+			},
+			eventsToGet: []SessionKey{
+				{"127.0.0.1", 0},
+				{"127.0.0.2", 0},
+			},
+			eventsExpected: []*bytes.Buffer{
+				bytes.NewBufferString("test-event-1"),
+				bytes.NewBufferString("test-event-1"),
+			},
+			getDelay:         1 * time.Second,
+			cleanInterval:    2 * time.Second,
+			removalThreshold: 2 * time.Second,
+		},
+		{
+			name: "Add and GetAndRemove same sessions with cache hit",
+			eventsToAdd: []testEvent{
+				{SessionKey{"127.0.0.1", 0}, bytes.NewBufferString("test-event-1")},
+				{SessionKey{"127.0.0.1", 0}, bytes.NewBufferString("test-event-1")},
+			},
+			eventsToGet: []SessionKey{
+				{"127.0.0.1", 0},
+			},
+			eventsExpected: []*bytes.Buffer{
+				bytes.NewBufferString("test-event-1"),
+				bytes.NewBufferString("test-event-1"),
+			},
+			getDelay:         1 * time.Second,
+			cleanInterval:    2 * time.Second,
+			removalThreshold: 2 * time.Second,
+		},
+		{
+			name: "Add and GetAndRemove with cache miss",
+			eventsToAdd: []testEvent{
+				{SessionKey{"127.0.0.1", 0}, bytes.NewBufferString("test-event-1")},
+				{SessionKey{"127.0.0.2", 0}, bytes.NewBufferString("test-event-1")},
+			},
+			eventsToGet: []SessionKey{
+				{"127.0.0.1", 0},
+				{"127.0.0.2", 0},
+			},
+			eventsExpected:   []*bytes.Buffer(nil),
+			getDelay:         2 * time.Second,
+			cleanInterval:    1 * time.Second,
+			removalThreshold: 1 * time.Second,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, cancelFunc := context.WithCancel(context.Background())
+			defer cancelFunc()
+			cache := newPendingTemplatesCache()
+			cache.start(ctx.Done(), tt.cleanInterval, tt.removalThreshold)
```

Comment:
maybe add a test for stopping the cache

Reply:
sure I am gonna add the `wait` inside the existing lru unit-tests 🙂

Reply:
I added the `wait` inside the lru unit-tests to guarantee that cache-related goroutine exits when ctx is Done  🙂 Feel free to unresolve if you spot anything else